qemu/tests/qemu_guest_agent.py

```diff
@@ -2543,13 +2544,88 @@ def gagent_check_fsinfo(self, test, params, env):
         self._open_session_list.append(session)
         serial_num = params["blk_extra_params_image1"].split("=")[1]
 
+        def get_guest_disk_usage(mountpoint):
+            '''
+            Send cmd in guest to get disk usage.
+            :param mountpoint: the mountpoint of filesystem
+            '''
+            cmd_get_disk_usage = params["cmd_get_disk_usage"] % mountpoint
+            disk_usageinfo_guest = session.cmd(cmd_get_disk_usage).strip().split()
+            disk_total_guest = int(disk_usageinfo_guest[0])
+            if params["os_type"] == "windows":
+                # Just can get total and freespace disk usage from windows.
+                disk_freespace_guest = int(disk_usageinfo_guest[1])
+                disk_used_guest = int(disk_total_guest - disk_freespace_guest)
+            else:
+                disk_used_guest = int(disk_usageinfo_guest[1])
+            return (disk_total_guest, disk_used_guest)
+
+        def check_diskusage_qga_guest(mount_point):
+            '''
+            Contrast the disk usage from guest and qga,
+            This function calls the previous function:'get_guest_disk_usage'.
+            :param mount_point: the mountpoint of filesystem
+            '''
+            disk_usage_guest = get_guest_disk_usage(mount_point)
+            disk_total_qga = int(fs["total-bytes"])
+            disk_used_qga = int(fs["used-bytes"])
+            if (disk_total_qga != disk_usage_guest[0] or
+                    disk_used_qga != disk_usage_guest[1]):
+                test.fail("File System Total bytes or Used bytes doesn't match.\n"
+                          "total_bytes from qga_cmd is %d."
+                          "total_bytes from guest is %d.\n"
+                          "used_bytes from qga_cmd is %d."
+                          "used_bytes from guest is %d."
+                          % (disk_total_qga, disk_usage_guest[0],
+                             disk_used_qga, disk_usage_guest[1]))
+            else:
```

Comment:
There is no need to add this else as we can check them in other logs and it's not the checkpoint.